Block Hooks: Remove feature gate and replace with woocommerce_hooked_blocks_version option

tjcafferkey opened the pull request:
### Changes proposed in this Pull Request:

We attempted to remove the feature gate for our hooked blocks in https://github.com/woocommerce/woocommerce/pull/45581 but ended up having to revert this change in https://github.com/woocommerce/woocommerce/pull/46935 due to the impact it had on existing stores where our hooked blocks suddenly appeared after updating WooCommerce on their site. Some of the challenges faced were:

1. Incompatible Designs
2. Incompatibility with Donation-Only Sites
3. Conflict with Existing Solutions
4. Unexpected Behaviour on Existing Sites

This PR introduces a new approach to block insertion. We introduce a new database option: `woocommerce_hooked_blocks_version`

1. **For new stores**: Set to the newly installed WooCommerce version (`9.2.0`)
2. **For existing stores with an approved block theme**: Set to `8.4.0` if they met previous feature gate conditions (e.g. had an approved theme).
3. **For existing stores without an approved block theme**: Don't set anything, preventing blocks getting hooked.
4. **For existing stores with a classic theme**: Set to their current WC version (`9.2.0` or above) once they switch to a block theme.

In addition, each hooked block is assigned a release version (existing hooked blocks use `8.4.0`)

Auto-insertion logic compares the block's version with `woocommerce_hooked_blocks_version` and only auto-inserts if block version is equal to it or less than.

This ensures we respect existing store designs while allowing auto-insertion for new stores.

### Existing merchants point of view
_"I want the introduction of new features to respect my existing store design and functionality so that I can maintain a consistent user experience for my customers and avoid unexpected changes"_

### Related

For similar future feature gating I have introduced a new DB option, whereby when a user installs WC it stores their initially installed version to the database https://github.com/woocommerce/woocommerce/pull/49139/ but since block hooks (v8.4) predates this new change (v9.2) we're unable to use it.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Important Note**: This change relies on a database migration. If the store has an approved theme and updates WooCommerce but does not run the migrations, the hooked blocks will stop being inserted until they do (unless they've persisted the header template part with the hooked blocks present). This is typically only an issue on self-hosted stores but I believe most hosts have a fix in place to make [this value](https://github.com/woocommerce/woocommerce/blob/c269a86efc004c1320c74af0c58613e2382a3836/plugins/woocommerce/includes/class-wc-install.php#L669) `true` similar to what [Calypso does](https://github.com/Automattic/wc-calypso-bridge/blob/d2c7a3cdfe3bcb5e21a29cb32afab034714608e6/includes/class-wc-calypso-bridge-setup.php#L84) meaning when an update happens (either manually or automatically) the migrations are run automatically too.

[woocommerce.zip](https://github.com/user-attachments/files/16232272/woocommerce.zip)

**New install**

1. Checkout this branch and run `pnpm run build:zip` to build the plugin or download the zip above.
2. Create a JN site (without WC) and install a plugin called [SQL Buddy](https://wordpress.org/plugins/sql-buddy/)
5. Ensure your site is using theme TT4
6. Go to Tools > SQL Buddy and within the `wp_options` table look for an option `woocommerce_hooked_blocks_version`, it should _not_ be present at this point.
7. Install the `woocommerce.zip` file you built/downloaded in step 1 and install it on your JN site.
8. Go back to Tools > SQL Buddy and within the `wp_options` table look for an option `woocommerce_hooked_blocks_version`, it should  be set to `9.2.0`.
9. Check that with the TT4 theme that the Mini Cart and My Account block are getting inserted into the header on the frontend and in the editor

**Existing stores updating WC with an approved block theme**

_This is to continue inserting hooked blocks for stores that are already getting them because they have an approved theme_

1. Checkout this branch and run `pnpm run build:zip` to build the plugin or download the zip above.
2. Create a JN site and install a plugin called [SQL Buddy](https://wordpress.org/plugins/sql-buddy/) and the latest version of [WooCommerce](https://wordpress.org/plugins/woocommerce/).
3. Go to Tools > SQL Buddy and within the `wp_options` table look for an option `woocommerce_hooked_blocks_version`, it should _not_ be present at this point.
4. Install the `woocommerce.zip` file you built/downloaded in step 1 and install it on your JN site.
5. Once installed, go to `/wp-admin/plugins.php` and run the database update (seen in the screenshot below) to run the migrations.
6. Go to Tools > SQL Buddy once the migrations are complete, and within the `wp_options` table look for an option `woocommerce_hooked_blocks_version`, it should set to `8.4.0`. This is the earliest version, and will ensure any future block hooks won't be inserted on the merchants store.
7. Check that the Mini Cart and Customer Account blocks are being inserted on the frontend and in the editor.

![Screenshot 2024-07-11 at 09 02 02](https://github.com/woocommerce/woocommerce/assets/8639742/c65a632d-dd91-4d93-bebb-a4eeb9d21032)

**Existing stores updating WC with a block theme that is not approved**

1. Checkout this branch and run `pnpm run build:zip` to build the plugin or download the zip above.
2. Create a JN site and install a block theme which isn't an approved them, such as [Purena](https://en-gb.wordpress.org/themes/purena/)
3. Install a plugin called [SQL Buddy](https://wordpress.org/plugins/sql-buddy/) and the latest version of [WooCommerce](https://wordpress.org/plugins/woocommerce/).
4. Go to Tools > SQL Buddy and within the `wp_options` table look for an option `woocommerce_hooked_blocks_version`, it should _not_ be present at this point.
5. Install the `woocommerce.zip` file you built/downloaded in step 1 and install it on your JN site.
6. Once installed, go to `/wp-admin/plugins.php` and run the database update (seen in the screenshot below) to run the migrations.
10. Go to Tools > SQL Buddy once the migrations are complete, and within the `wp_options` table look for an option `woocommerce_hooked_blocks_version`, it should be set to `no` because you are using an unapproved block theme.
11. Check that no hooked blocks are appearing on the frontend or in the editor.
12. Now activate a classic theme (e.g. Storefront) and then switch back to TT4. Check that the option in the database is still set to `no`
13. Check that no hooked blocks are appearing on the frontend or in the editor.

![Screenshot 2024-07-11 at 09 02 02](https://github.com/woocommerce/woocommerce/assets/8639742/c65a632d-dd91-4d93-bebb-a4eeb9d21032)

**Existing stores that had a classic theme and then switch to a block theme**

1. Checkout this branch and run `pnpm run build:zip` to build the plugin or download the zip above.
2. Create a JN site and install a block theme which isn't an approved them, such as [Storefront](https://wordpress.org/themes/storefront/)
3. Install a plugin called [SQL Buddy](https://wordpress.org/plugins/sql-buddy/) and the latest version of [WooCommerce](https://wordpress.org/plugins/woocommerce/).
4. Go to Tools > SQL Buddy and within the `wp_options` table look for an option `woocommerce_hooked_blocks_version`, it should _not_ be present at this point.
5. Install the `woocommerce.zip` file you built/downloaded in step 1 and install it on your JN site.
6. Once installed, go to `/wp-admin/plugins.php` and run the database update (seen in the screenshot below) to run the migrations.
8. Go to Tools > SQL Buddy once the migrations are complete, and within the `wp_options` table look for an option `woocommerce_hooked_blocks_version`, it should _not_ be present because you are using an incompatible theme.
14. Switch to a block theme such as TT4.
15. Go to Tools > SQL Buddy once the migrations are complete, and within the `wp_options` table look for an option `woocommerce_hooked_blocks_version`, it should be set to `9.2.0` (your current version of WC).
16. Check that the Mini Cart and My Account block are appearing on the frontend and in the editor

![Screenshot 2024-07-11 at 09 02 02](https://github.com/woocommerce/woocommerce/assets/8639742/c65a632d-dd91-4d93-bebb-a4eeb9d21032)

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

tjcafferkey committed:
Remove the block hooks feature gate by getting and setting a the wc_hooked_blocks_version option and assigning versions to each hooked block

tjcafferkey committed:
Update apply_filters comment

tjcafferkey committed:
Add action for users switching themes that don't have an option set

tjcafferkey committed:
Update condition for current_theme_supports usage

tjcafferkey committed:
Update comment

tjcafferkey committed:
Unit tests

tjcafferkey committed:
Change unit test file name

tjcafferkey committed:
Update hooks for option update

tjcafferkey committed:
Changelog

tjcafferkey committed:
Add database tests for wc_hooked_blocks_version actions

tjcafferkey committed:
Update option name from wc_hooked_blocks_version to woocommerce_hooked_blocks_version

tjcafferkey committed:
Update option name in function comments

Aljullu commented on plugins/woocommerce/includes/wc-template-functions.php:
```diff
@@ -4019,3 +4019,43 @@ function wc_update_product_archive_title( $post_type_name, $post_type ) {
+function wc_set_hooked_blocks_version_on_theme_switch( $old_name, $old_theme ) {
+	$option_name = 'woocommerce_hooked_blocks_version';
+
+	if ( ! $old_theme->is_block_theme() && ( wc_current_theme_is_fse_theme() || current_theme_supports( 'block-template-parts' ) ) && ! get_option( $option_name ) ) {
```
Out of curiosity, why only when switching from a classic to a block theme and not when switching between block themes?

tjcafferkey replied:
Hmm good question. I kind of touched on this in the [comment](https://github.com/woocommerce/woocommerce/pull/49302#issuecomment-2222926972) below:

>if they change themes, and then change back to their original theme they would get all of the latest hooked blocks. Would you consider that a problem in your opinion?

Aljullu replied:
After our conversation yesterday in Slack, we decided that stores would be "locked" to the block hooks which were available at the time of installation, right? Then, can this function be removed? :thinking:

Aljullu commented on plugins/woocommerce/includes/wc-update-functions.php:
```diff
@@ -2729,3 +2729,30 @@ function wc_update_891_create_plugin_autoinstall_history_option() {
+	if ( $should_set_hooked_blocks_version && ! get_option( $option_name ) ) {
+		// Set 8.4.0 as the version for existing stores that are using a theme that supports the Block Hooks API.
+		// This will ensure that the Block Hooks API is enabled for these stores and works as expected.
+		// Existing stores that aren't running allowed these themes will not have the Block Hooks API enabled.
```
I think there is a typo in this comment: "aren't running allowed these themes".

tjcafferkey replied:
Good catch, updated in https://github.com/woocommerce/woocommerce/pull/49302/commits/63b133a926d12979fccf6e5170e340caea780d88

tjcafferkey committed:
Merge branch 'trunk' into remove/block-hooks-feature-gate

tjcafferkey committed:
Fix linting

Aljullu reviewed (commented):
Thanks for all the effort to make sure we properly handle existing stores! :raised_hands: Also thanks for writing several unit tests for this!

While reviewing this, I found a case that is not clear to me:

* The user installs WC at version 9.2 (for example).
* A new block using hooks is introduced in 9.3.
* The store switches themes in 9.4.

Should the block hook from 9.3 be enabled? In other words, should `woocommerce_hooked_blocks_version` be reset to the latest version every time we switch themes or it should stick to be the same value as `woocommerce_initial_install_version`?

tjcafferkey commented:
>In other words, should `woocommerce_hooked_blocks_version` be reset to the latest version every time we switch themes or it should stick to be the same value as `woocommerce_initial_install_version`?

Good question! Firstly `woocommerce_initial_installed_version` isn't used in this implementation at all, sorry I wasn't sure if that is clear.

Secondly if a user switches theme we only set the `woocommerce_hooked_blocks_version` if it hasn't already been set. In short, if a version already exists we _never_ update it. But having a rethink, maybe we should in this scenario since if they're changing themes maybe they should expect UI changes.

But as a result what this would also mean is that if they change themes, and then _change back_ to their original theme they would get all of the latest hooked blocks. Would you consider that a problem in your opinion?

tjcafferkey committed:
Update comment typo

tjcafferkey committed:
Add strict types to new files

tjcafferkey committed:
Merge branch 'trunk' into remove/block-hooks-feature-gate

tjcafferkey committed:
Merge branch 'trunk' into remove/block-hooks-feature-gate

Aljullu reviewed (commented):
> Good question! Firstly `woocommerce_initial_installed_version` isn't used in this implementation at all, sorry I wasn't sure if that is clear.

Yup, it was clear. :slightly_smiling_face: I mentioned because if I'm understanding it right, `woocommerce_initial_installed_version` and `woocommerce_hooked_blocks_version` will hold the same value for all new stores, right? Which makes me wonder... do we really need to save `woocommerce_hooked_blocks_version` for new stores? Not a big deal but this way we can remove `wc_set_hooked_blocks_version()`.

tjcafferkey commented:
>Which makes me wonder... do we really need to save woocommerce_hooked_blocks_version for new stores? Not a big deal but this way we can remove wc_set_hooked_blocks_version().

Currently there are some existing stores that get hooked blocks, but don't have the `woocommerce_initial_installed_version` option. We need to ensure these stores continue getting these blocks which is what [this change handles](https://github.com/woocommerce/woocommerce/pull/49302/files#diff-ef66873bc65699d07994f8ac9621c771df9f9e63ce74c94b70defac87aab52d1R2736).

The reason we have to use `woocommerce_hooked_blocks_version` is because on the migration I linked above we don't know what their `woocommerce_initial_installed_version` value would be.

Unless I am thinking about this wrong?

tjcafferkey committed:
Update tests

tjcafferkey committed:
Move unit tests for block hooks

tjcafferkey committed:
Set option to "no" for sites that dont have an approved block theme

tjcafferkey commented:
>After our conversation yesterday in Slack, we decided that stores would be "locked" to the block hooks which were available at the time of installation, right?

As per our conversation with is what we agreed (please correct if I am wrong 😅 )

* Site switches from classic theme > block theme for the first time. **Insert all latest block hooks version**
* Site switches from block theme > classic theme > block theme. **Respect their existing block hooks version**
* Site switches from block theme > block theme. **Respect their existing block hooks version**

You can find the change that achieves this here https://github.com/woocommerce/woocommerce/pull/49302/commits/befa00248b758e4b5117ef499941e9399b87de3a

>User switches from block theme > classic theme > block theme. **Respect their existing block hooks**

In order to account for this we updated the migration to set a `no` as the value if the user is running a block theme that _isn't_ approved. This way, if there is no option set at all (e.g. `false`) then we can assume the user never had a block theme active and its the first time they have switched from a classic theme > block theme

I've added some unit tests to to support these scenarios in [wc-update-functions-test.php](https://github.com/woocommerce/woocommerce/pull/49302/commits/befa00248b758e4b5117ef499941e9399b87de3a#diff-589fcdd82b13562c9deced6d75368c0c9572236473d534a5080b7d9252e86ccb). Please let me know if there is anything I've missed here or if something is unclear.

It's also worth noting that every time we interact with the option we always use `add_option` rather than `update_option` to avoid overwriting the value.

Aljullu commented on plugins/woocommerce/includes/class-woocommerce.php:
```diff
@@ -307,6 +307,7 @@ private function init_hooks() {
 		self::add_action( 'rest_api_init', array( $this, 'register_wp_admin_settings' ) );
 		add_action( 'woocommerce_installed', array( $this, 'add_woocommerce_remote_variant' ) );
 		add_action( 'woocommerce_updated', array( $this, 'add_woocommerce_remote_variant' ) );
+		add_action( 'woocommerce_newly_installed', 'wc_set_hooked_blocks_version', 10 );
```
Out of curiosity, why defining `wc_set_hooked_blocks_version` in `wc-template-functions.php` instead of here? I see all other hooks are methods of this class.

tjcafferkey replied:
Hmm, not every callback used in these hooks is present on that class as a method though (e.g. some actions). I guess my reasoning was that this class is responsible for the setup of some hooks, but the definition of this hook was too specific to belong as a method on the class itself. Felt like I was polluting it.

Aljullu replied:
Oh, you're right, some callbacks are from other classes. Yeah, no need to change anything here. :+1:

Aljullu commented on plugins/woocommerce/includes/wc-update-functions.php:
```diff
@@ -2730,6 +2730,39 @@ function wc_update_910_add_launch_your_store_tour_option() {
+	}
+
+	$option_name       = 'woocommerce_hooked_blocks_version';
+	$option_value      = get_option( $option_name );
```
Nit: given that in the if clause below we check whether `$option_value === false` in both cases, what do you think about making it an early return? Ie:

```PHP
	...
	$option_value = get_option( $option_name );
	if ( false !== $option_value ) {
		return;
	}
	...
```

tjcafferkey replied:
We could do but it feels redundant don't you think? Given that this is the migration we can pretty much be sure that the option doesn't exist since we only start introducing it to new stores (which wont run this migration) in this PR also?

tjcafferkey replied:
But then again, I do the `false` check in the conditions 🤔

tjcafferkey replied:
Done in https://github.com/woocommerce/woocommerce/pull/49302/commits/00cf4adb729d118621073cfc9ec988f844d2b9db

Aljullu commented on plugins/woocommerce/includes/wc-update-functions.php:
```diff
@@ -2730,6 +2730,39 @@ function wc_update_910_add_launch_your_store_tour_option() {
+		add_option( $option_name, '8.4.0' );
+	} elseif ( ! $should_set_hooked_blocks_version && false === $option_value ) {
+		// For block themes that aren't approved themes set this option to "no" so we can disable the Block Hooks API.
+		// This will allow us to set a version when users switch from a classic theme to a block theme.
```
I'm not sure to understand this comment, aren't we setting `no` for the opposite reason? In other words, we are setting `no` so the option is defined so when switching from a classic theme to a block theme we won't update the option.

I might be getting confused, though. :sweat_smile:

tjcafferkey replied:
We set "no" for unapproved block themes so when they go from block theme > classic theme > back to block theme we don't update it.

However, for classic themes > block theme for the first time it means we can set the option value to be the current WC version as we outlined in this comment https://github.com/woocommerce/woocommerce/pull/49302#issuecomment-2226163542

Aljullu replied:
Hmmm, but here ([line 2768](https://github.com/woocommerce/woocommerce/pull/49302/files#diff-ef66873bc65699d07994f8ac9621c771df9f9e63ce74c94b70defac87aab52d1R2768)) we are setting the option to `no`. That means it won't be overridden when switching from a classic theme to a block theme (according to the check [here](https://github.com/woocommerce/woocommerce/pull/49302/files#diff-d49488c34017d273db858b7b6eff58f22d95310991d26638e75793c6a1e98678R4060)), no? Thus, contradicting this part of the comment: `This will allow us to set a version when users switch from a classic theme to a block theme`.

I might be miunsdertanding this comment, but I think it doesn't match what's happening in this part of the `elseif` clause? :thinking:

tjcafferkey replied:
The first line of the function [checks if the user is _currently_ running a classic theme](https://github.com/woocommerce/woocommerce/pull/49302/files#diff-ef66873bc65699d07994f8ac9621c771df9f9e63ce74c94b70defac87aab52d1R2737-R2739) and if they are we don't set an option _at all_ which means when they [switch from classic > block theme we can set one then](https://github.com/woocommerce/woocommerce/pull/49302/files#diff-d49488c34017d273db858b7b6eff58f22d95310991d26638e75793c6a1e98678R4055) if they haven't already got the option we can set it there as the latest value.

IF they have already got the option it is because when the migration ran on the update they were running a block theme and thus we respect the versioning value.

Aljullu commented on plugins/woocommerce/src/Blocks/Utils/BlockHooksTrait.php:
```diff
@@ -18,31 +18,34 @@ trait BlockHooksTrait {
-		static $active_theme_name = null;
-		if ( is_null( $active_theme_name ) ) {
-			$active_theme_name = wp_get_theme()->get( 'Name' );
+		// If the block has no hook placements, return early.
+		if ( ! isset( $this->hooked_block_placements ) || empty( $this->hooked_block_placements ) ) {
+			return $hooked_blocks;
 		}
```
Nit: wouldn't it be more performant to do this check _before_ checking the `$block_hooks_version`? Even if it's cached, checking `$block_hooks_version` requires one `get_option()`, while checking `$this->hooked_block_placements` is more performant, no? :thinking:

tjcafferkey replied:
Makes sense!

tjcafferkey replied:
Done in https://github.com/woocommerce/woocommerce/pull/49302/commits/56332e1558226714519990a444b2e035d845cd97

Aljullu commented on plugins/woocommerce/tests/php/src/Blocks/BlockTypes/BlockHooksTests.php:
```diff
@@ -0,0 +1,85 @@
+	}
+
+	/**
+	 * Test block does not get hooked with incorrect version
```
What do you think about changing "incorrect version" to "lower version" or something similar? When reading the test description I thought it would set the option to a wrong format instead of a correct format but lower number.

tjcafferkey replied:
Nice catch I agree

tjcafferkey replied:
Done in https://github.com/woocommerce/woocommerce/pull/49302/commits/2dec5c4bdba14cdd88dfd6e19d87ae2f9063cca8

Aljullu commented on plugins/woocommerce/tests/php/src/Blocks/BlockTypes/BlockHooksTests.php:
```diff
@@ -0,0 +1,85 @@
+		$this->assertNotContains(
+			'woocommerce/test-block',
+			$hooked_block_types,
+			'Hooked block shouldnt be added unless a version is set'
```
Small typo:

```suggestion
			'Hooked block shouldn't be added unless a version is set'
```

tjcafferkey replied:
Nice catch done in https://github.com/woocommerce/woocommerce/pull/49302/commits/14363e03180ad44e45cbf3b9d0403ed6a899a719

Aljullu reviewed (approved):
Awesome job here, @tjcafferkey! :clap: I just tested and all use cases work as expected. I also took a look at the code and everything looks good. I left some small suggestions inline, but pre-approving anyway.

Besides that, a couple of comments about the testing notes:

* Maybe we should clarify that this step doesn't need to be done when testing the release, no?

> Checkout this branch and run pnpm run build:zip to build the plugin or download the zip above.

* And I think this step should say "classic theme", right? :thinking: 

> Create a JN site and install a block theme which isn't an approved them, such as [Storefront](https://wordpress.org/themes/storefront/)

tjcafferkey committed:
Return early if WC option already exists

tjcafferkey committed:
Reorder conditions for performance reasons

tjcafferkey committed:
Change incorrect to lower

tjcafferkey committed:
Fix typo

Aljullu commented on plugins/woocommerce/includes/wc-update-functions.php:
```diff
@@ -2730,6 +2730,45 @@ function wc_update_910_add_launch_your_store_tour_option() {
+	$theme_include_list               = apply_filters( 'woocommerce_hooked_blocks_theme_include_list', array( 'Twenty Twenty-Four', 'Twenty Twenty-Three', 'Twenty Twenty-Two', 'Tsubaki', 'Zaino', 'Thriving Artist', 'Amulet', 'Tazza' ) );
+	$should_set_hooked_blocks_version = in_array( $active_theme_name, $theme_include_list, true );
+
+	if ( $should_set_hooked_blocks_version && false === $option_value ) {
+		// Set 8.4.0 as the version for existing stores that are using a theme that supports the Block Hooks API.
+		// This will ensure that the Block Hooks API is enabled for these stores and works as expected.
+		// Existing stores that aren't running approved block themes will not have the Block Hooks API enabled.
+		add_option( $option_name, '8.4.0' );
+	} elseif ( ! $should_set_hooked_blocks_version && false === $option_value ) {
```
It should be possible to remove `false === $option_value` from these checks now, no?

```suggestion
	if ( $should_set_hooked_blocks_version ) {
		// Set 8.4.0 as the version for existing stores that are using a theme that supports the Block Hooks API.
		// This will ensure that the Block Hooks API is enabled for these stores and works as expected.
		// Existing stores that aren't running approved block themes will not have the Block Hooks API enabled.
		add_option( $option_name, '8.4.0' );
	} elseif ( ! $should_set_hooked_blocks_version ) {
```

tjcafferkey replied:
Done in https://github.com/woocommerce/woocommerce/pull/49302/commits/8f583ee6714d2f839e832f28756d5402bef7c45a

tjcafferkey committed:
Remove unncessary false check and add additional unit tests

tjcafferkey committed:
Update comment
